common/cache/namespaceCache.go

```diff
@@ -399,12 +400,11 @@ func (c *namespaceCache) refreshLoop() {
 			return
 		case <-timer.C:
 			for err := c.refreshNamespaces(); err != nil; err = c.refreshNamespaces() {
-				select {
-				case <-c.shutdownChan:
+				c.logger.Error("Error refreshing namespace cache", tag.Error(err))
+				time.Sleep(NamespaceCacheRefreshFailureRetryInterval)
+
+				if _, opened := <-c.shutdownChan; !opened {
```

Comment:
Exit immediately if error occured and shutdown channel is closed. For some reason (I don't quite understand why) it did stuck here before.